### DISCUSSION TITLE
list forbidden attributes

### DISCUSSION THREAD

**Discussion by mlarcher** [apostrophecms/sanitize-html#587] [Q&A] [answered]
Is there a way to blacklist some attributes (like `style` for example) on all tags ?

I can see a whiteList with `allowedAttributes` but it would come in more handy to have a `forbiddenAttributes` that would remove those attributes, instead of listing all others, which seems close to impossible...

**Comment by boutell** [comment, marked as answer]
I think it is the other way around: there are too many attributes out there that can embed JavaScript, etc. for a "forbidden list" to ever really be safe. Or it could be safe for a day and then a new attribute pops up in some browsers. Always use an allowed list.

**Reply by mlarcher** [reply]
Right, but my concern is not really about safety here : even thoughI am indeed using the lib mostly for safety, I also happen to want to prevent `style` attribute in the provided html. 
Why wouldn't sanitize-html provide this kind of feature, considering it already does most of the job ? I can't find any lib that would handle that use case reliably, so I'd be very interested by a way to do so with sanitiz-html, be it with a custom filter of sorts. Anything possible in that area ?

**Comment by boutell** [comment]
I see your point. When combined with an allow list, a forbid list is useful.

That would make a good PR.

On Wed, Dec 7, 2022 at 5:07 PM Matthieu Larcher ***@***.***>
wrote:

> Right, but my concern is not really about safety here : even thoughI am
> indeed using the lib mostly for safety, I also happen to want to prevent
> style attribute in the provided html.
> Why wouldn't sanitize-html provide this kind of feature, considering it
> already does most of the job ? I can't find any lib that would handle that
> use case reliably, so I'd be very interested by a way to do so with
> sanitiz-html, be it with a custom filter of sorts. Anything possible in
> that area ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/sanitize-html/discussions/587#discussioncomment-4337669>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27J7SKGZADIKZRRSBDTWMEDCZANCNFSM6AAAAAASWUROME>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his
